system-tests/lib/cre/workflow/workflow.go

```diff
@@ -77,6 +81,86 @@ func RegisterWithCRECLI(input cretypes.RegisterWorkflowWithCRECLIInput) error {
 	return nil
 }
 
+func validateRegisterWorkflowInput(input cretypes.ManageWorkflowWithCRECLIInput) error {
+	if input.ShouldCompileNewWorkflow && input.NewWorkflow == nil {
+		return errors.New("NewWorkflow is required when ShouldCompileNewWorkflow is true")
+	}
+	if !input.ShouldCompileNewWorkflow && input.ExistingWorkflow == nil {
+		return errors.New("ExistingWorkflow is required when ShouldCompileNewWorkflow is false")
+	}
+	if input.NewWorkflow != nil && input.NewWorkflow.FolderLocation == "" {
+		return errors.New("WorkflowFolderLocation is required when ShouldCompileNewWorkflow is true")
+	}
+	if input.NewWorkflow != nil && input.NewWorkflow.WorkflowFileName == "" {
+		return errors.New("WorkflowFileName is required when ShouldCompileNewWorkflow is true")
+	}
+	if input.ExistingWorkflow != nil && input.ExistingWorkflow.BinaryURL == "" {
+		return errors.New("BinaryURL is required when ShouldCompileNewWorkflow is false")
+	}
+
+	return nil
+}
+
+func PauseWithCRECLI(input cretypes.ManageWorkflowWithCRECLIInput) error {
+	if valErr := input.Validate(); valErr != nil {
+		return errors.Wrap(valErr, "failed to validate WorkflowInput")
+	}
+
+	// This env var is required by the CRE CLI
+	pkErr := os.Setenv("CRE_ETH_PRIVATE_KEY", input.CRECLIPrivateKey)
+	if pkErr != nil {
+		return errors.Wrap(pkErr, "failed to set CRE_ETH_PRIVATE_KEY")
+	}
+
+	// This env var is required by the CRE CLI
+	profileErr := os.Setenv("CRE_PROFILE", input.CRECLIProfile)
+	if profileErr != nil {
+		return errors.Wrap(pkErr, "failed to set CRE_PROFILE")
+	}
+
+	creCLIWorkflowSettingsFile, err := libcrecli.PrepareCRECLIWorkflowSettingsFile(input.CRECLIProfile, input.WorkflowOwnerAddress, input.WorkflowName)
+	if err != nil {
+		return err
+	}
```

Comment:
would it make sense to extract all these lines above to a separate function? Seems that it is the same for all operations.

Also, maybe `creCLIWorkflowSettingsFile` could be passed in input, so we don't create it over and over again?